src/pygenesis/find.py

```diff
@@ -0,0 +1,269 @@
+"""Implements find endpoint to retrieve results based on query"""
+
+import pandas as pd
+
+from pygenesis.http_helper import get_response_from_endpoint
+
+pd.set_option("max_colwidth", None)
+pd.set_option("expand_frame_repr", False)
+
+
+class Results:
+    """
+    A class representing the result object of variables, statistics, cubes and tables.
+
+    Attributes:
+        df (pd.DataFrame): The DataFrame that contains the data.
+        category (str): Category of the result. E.g. "tables", "cubes".
+
+    Methods:
+        get_code(): Gets code based on the index of the object.
+        get_metadata(): Gets metadata based on the index of the object.
+    """
+
+    def __init__(self, result: pd.DataFrame, category: str):
+        """
+        Class that contains the results of a find query.
+
+        Args:
+            DataFrame (pd.DataFrame): Result of a search query.
+            category (str): Category of the result. E.g. "tables", "cubes"
+        """
+        self.df = result
+        self.category = category
+
+    def __repr__(self):
+        return self.df.to_markdown()
+
+    def __str__(self):
+        return self.df.to_markdown()
+
+    def __len__(self):
+        if len(self.df) > 0:
+            return len(self.df)
+        else:
+            return 0
+
+    def get_code(self, index: list):
+        """
+        Returns the for a given list of tables
+
+        Args:
+             index (list): A list that contains the indices from the results objects. This is not the object code."
+        Returns:
+             table codes (list): Contains the corresponding tables codes.
+        """
+
+        codes = self.df.iloc[index]["Code"]
+        return list(codes)
+
+    def get_metadata(self, index: list):
+        """
+        Prints meta data for a given list of tables.
+
+        Args:
+              index (list): A list that contains the indices from the results objects. This is not the object code."
+        Returns:
+              prints meta data for all indices.
+        """
+        codes = self.df.iloc[index]["Code"]
+
+        for code, ix in zip(codes, index):
+            response = self._get_metadata_results(self.category[0:-1], code)
```

Comment:
Warum muss hier aus category das letzte Zeichen abgetrennt werden? Das ist mega hacky und nicht nachvollziehbar. Dann doch bitte in category auch singular Speichern, also "table" und dort, wo man plural braucht ein "s" dranhängen?

Reply:
Beim parameter zum find enpunkt wird der Plural verwendet, bei der Url der Metadata Methode, der Singular. Das ist ein wenig unschön in der API umgesetzt, deswegen die Verwirrung. Haben das im Docstring nochmal klarer definiert.